orm/go.mod

```diff
@@ -9,22 +9,29 @@ require (
 	github.com/golang/mock v1.6.0
 	github.com/google/go-cmp v0.5.7
 	github.com/iancoleman/strcase v0.2.0
+	github.com/regen-network/gocuke v0.6.1
```

Comment:
Please don't introduce new testing frameworks to the project.

Reply:
Why?

Reply:
There's no reason for it. Everything here is equally well expressed as conventional table-based Go unit tests.

Reply:
Each new dependency added to `go.mod` must be default deny. The onus is on the author to justify its value.

Reply:
The goal is readability. We've decided internally to try this type of testable documentation. If it doesn't provide value, we won't continue with it

Reply:
Sorry, strong -1. Standard Go tests are perfectly readable, and understood by all Go programmers by definition. This introduces a brand new DSL which maintainers would have to learn and work within, and provides no clear benefit versus the alternative.

Reply:
Testing DSLs like Cucumber can make sense in dynamic and monkey-patchable languages like Ruby. They don't make sense in Go.

Reply:
Great thanks for your opinion. Several of us have decided to try this approach. We are not committed to anything yet as an overall strategy but we have found this approach useful in some modules. The maintainers will discuss more and decide where we find this useful or not useful

Reply:
I would also add that readability of tests has been flagged as an issue and one of the big issues I have often seen with SDK code is specifying too many implicit behavioral details only in go code and tests. These are state machines that must always be deterministic and must always have well specified behavior, and must be understandable by non go developers and sometimes implemented in other languages

Reply:
"Acceptance tests" are valuable! But acceptance tests can be written in Go. It's not obvious what value is delivered by a testing DSL.

> [these tests]  must be understandable by non go developers and sometimes implemented in other languages

The Cosmos SDK is written in Go. Anyone who contributes to it must understand Go by definition. It cannot be a goal that tests, or any other part of the SDK, can be understandable, or written in, languages other than Go.

Reply:
> It's not true that test written in a testing DSL are "more readable" than tests written in the underlying language by construction.

That's definitely debatable.

> It cannot be a goal that tests, or any other part of the SDK, should be understandable, or written in, languages other than Go.

Actually it is very much a goal that significant parts of the SDK are written in another language already - specifically protobuf. And it has been a goal of the SDK for over 2 years now. Previously there was a goal to do as much as possible in golang with amino and most core developers as well as client developers were unhappy with that stance as it caused LOTS of downstream problems and decided to adopt protobuf and use it as much as possible. I'm sorry but it is really not your role to state what is or is not a goal of the SDK. I am not trying to shut down discussion of this topic out of hand because I acknowledge that this is a common good, open source project with many stakeholders and that discussion is important and helpful. Yet, just simply stating that something must or must not be a goal of the project is not the way to have a constructive discussion. I have not seen a lot of people (actually anyone) begging to go back to amino and lots of newer tools are being built on the value provide by a non-golang specification tool, so I think I can confidently say that the SDK having a non-goal of specifying as much as possible in golang is inline with the common good interest of many stakeholders.

Furthermore, this SDK is not about golang. It is written in golang, but specifies state machines and protocols that are used by many many people who have no golang knowledge. And it is a long term goal that some of these state machines will have spec client implementations in other languages such as Rust and Typescript.

So from my perspective this approach is inline with the larger goals of the project and the other maintainers who are involved in this work.

Reply:
@peterbourgon I'm a bit confused why you're singling out a new package dependency as a first place to start engaging in the SDK. Maybe I've missed your other contributions here, but it feels a bit out of place for you to enter into this PR with such a specific complaint without having a longer and more established relationship with the SDK project and the contributors.

I understand why one may have the perspective you're voicing here, but imo these kinds of decisions should be left to the maintainers of the project.  I'd like to ask that you refrain from further pushing on this issue here, and let the SDK devs explore the value of integrating cucumber acceptance tests into the development process.

If there are concrete ways in which adding such a framework is affecting you and your work (either as a developer building on the SDK, or as a potential contributor to the SDK), I'd encourage you to file an issue so this PR can move forward with reviews from the existing core developers.

Reply:
Thank you for everyones input. @peterbourgon thank you for your input, you have already made insightful contributions in helping review code. We would like to move forward with this dependency, if in the future we deem this dependency to be an issue or not needed anymore we will gladly remove it, but after discussions on testing procedures we have come to the conclusion that this may be an easier path to involve more people in our writing of tests. 

We value your reviews and your work in the cosmos ecosystem and look forward to working with you in the future 😄

Reply:
I expressed this 1:1 to @aaronc, I also have doubts about go-cuke.

Aaron convinced me that go-cuke is more readable, and definitely has value for non-dev people (e.g. product people) wanting to know what a module does. I was less sure about value for developers, but I'm fine to try it and see how it fits in.

As for adding go-cuke today in this PR, there are a couple of questions that I'd like to see answered:
1. https://github.com/regen-network/gocuke is a repo of 62 (non-reviewed?) commits by Aaron. If we add this to the SDK, we need to reduce the bus factor first audit it (or at least understand it).
2. The acceptance tests should test that code matches spec, where "spec" is generally an agreed-upon document in English language. We don't have this doc yet (Marko and I will try to work with Aaron on that), it seems now that these tests almost are the spec itself, is that what BDD precognizes?

Reply:
It might be useful to get an ADR on testing up and which way things are going. We all understand there is an issue with testing in the cosmos-sdk that leads to messed up ci pipelines and not having full coverage. 

I thought there was agreement on this from the Regen side, but seems there are conflicting opinions.

Reply:
I have clearly misunderstood the purpose of PRs in this project. My apologies. I'll refrain from commenting or contributing any further.

Reply:
> 1. https://github.com/regen-network/gocuke is a repo of 62 (non-reviewed?) commits by Aaron. If we add this to the SDK, we need to reduce the bus factor first audit it (or at least understand it).

Folks on the regen ledger side have looked into this more and that is where our devs are more aligned around this approach @marbar3778. We can prepare something a bit more formal if that helps. Although the reality is we don't usually do too thorough of an audit of 3rd party test dependencies as is...

> 2. The acceptance tests should test that code matches spec, where "spec" is generally an agreed-upon document in English language. We don't have this doc yet for ORM (Marko and I will try to work with Aaron on that), it seems now that these tests almost are the spec itself, is that what BDD precognizes?
> 

That's the point. There is no English language spec. That's precisely what this introduces